library/src/test/java/com/yayandroid/locationmanager/mocks/MockSimpleTask.java

```diff
@@ -0,0 +1,143 @@
+package com.yayandroid.locationmanager.mocks;
+
+import android.app.Activity;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.RuntimeExecutionException;
+import com.google.android.gms.tasks.Task;
+
+import java.util.concurrent.Executor;
+
+public class MockSimpleTask<TResult> extends Task<TResult> {
```

Comment:
Consider implementing this to reflect what you expect it to do, as in
`new MockSimpleTask<>(((Location) null))` is not clear what is your intention, obviously I know that you mean it to be successful now since I check the implementation, but you can build the API as following
```
MockSimpleTask mockSimpleTask = new MockSimpleTask<Location>();
mockSimpleTask.success(resultToVerify);
// verify expected action
verify(mockedSource).requestLocationUpdate();
```

Reply:
...and to achieve this, you might need to keep list of listeners and call them when requested.

```
public class MockSimpleTask<TResult> extends Task<TResult> {

    private final List<OnSuccessListener<? super TResult>> onSuccessListeners = new ArrayList<>();

    public void success(TResult result) {
         for(OnSuccessListener<? super TResult> onSuccessListener: onSuccessListeners) {
            onSuccessListener.onSuccess(var1);
        }
    }

    @NonNull
    public Task<TResult> addOnSuccessListener(@NonNull OnSuccessListener<? super TResult> var1) {
        onSuccessListeners.add(var1);
        return this;
    }

}
```

Reply:
Done

Reply:
Thank you for the changes